tests/PHPStan/Analyser/nsrt/bug-11311.php

```diff
@@ -0,0 +1,19 @@
+<?php // lint >= 7.4
+
+namespace Bug11311;
+
+use function PHPStan\Testing\assertType;
+
+function doFoo(string $s) {
+	if (1 === preg_match('/(?<major>\d+)\.(?<minor>\d+)(?:\.(?<patch>\d+))?/', $s, $matches, PREG_UNMATCHED_AS_NULL)) {
+
+		assertType('array{0: string, major: string, 1: string, minor: string, 2: string, patch: string|null, 3: string|null}', $matches);
```

Comment:
another fix this PR provides: a non-optional group cannot be `null` even when `PREG_UNMATCHED_AS_NULL` is given